bundles/org.openhab.binding.awattar/src/main/java/org/openhab/binding/awattar/internal/handler/AwattarBestPriceHandler.java

```diff
@@ -163,7 +162,7 @@ public void refreshChannel(ChannelUID channelUID) {
         long diff;
         switch (channelId) {
             case CHANNEL_ACTIVE:
-                state = OnOffType.from(result.isActive());
+                state = OnOffType.from(result.isActive(getNow(zoneId).toInstant()));
```

Comment:
Here and other places below: This can be simplified, there is no reason to go through the trouble of getting a `ZonedDateTime` just to discard to time-zone again when converting back to `Instant`:
```suggestion
                state = OnOffType.from(result.isActive(Instant.now()));
```

Reply:
The main reason for this change is to access the instant in the utests for stubbing. 
I did not find a way with the available framework to stub the const `Instant.now()` method.

Reply:
hwo about this chnage in https://github.com/openhab/openhab-addons/pull/17752/commits/622cfc0e48dd6de9815d13806e88edd7bed6dd2a

would this work for you?

Reply:
You might also be able to provide a [Clock](https://igm.univ-mlv.fr/~juge/javadoc-17/api/java.base/java/time/Clock.html) object for mocking this. See for example:
https://github.com/openhab/openhab-addons/blob/98d257982c0c37db2712f2f8f0fc33e703700df7/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/PriceListParser.java#L41-L59

and:
https://github.com/openhab/openhab-addons/blob/98d257982c0c37db2712f2f8f0fc33e703700df7/bundles/org.openhab.binding.energidataservice/src/test/java/org/openhab/binding/energidataservice/internal/PriceListParserTest.java#L67-L79

Reply:
Ijust had a look and it seems the example you send me implements a Constructor where the clock object is initialized exactly for the reason of testing.
In the production code, only the constructor without the parameter is used.

I can change the code, but i think there is not much difference between the two approaches.

Reply:
i have worked on it and changed most of the code to a clock object to have the same look and feel.